### DISCUSSION TITLE
Better support of Socket.io server

### DISCUSSION THREAD

**Discussion by datteroandrea** [vercel/next.js#47742] [Ideas]
### Goals

1. Improve the support of Socket.io server
2. Allow hot reload on Socket.io changes

### Background

I have been working with Socket.io and NextJS and all I can say is that using these 2 technologies together it's very hard.

The current approach as shown in this article [here](https://blog.logrocket.com/implementing-websocket-communication-next-js/) makes the server code of the socket.io very verbose and almost unmaintainable, there should be better ways of implementing a Socket.io server in NextJS.

Also using the current approach prevents hot-reload which means that you need to restart the entire app everytime you make changes to the socket.io server and this makes the development really annoying.

### Proposal

What about having a Server class that can declare methods decorated with some annotation that identify the event triggered by the client to be handled by the server? I am also willing to help implement this feature.

**Comment by gabrielsalvador** [comment]
I second this.

**Reply by geneshairzan** [reply]
third to this.... 

my issue relate to maintaining socket state at client side. 
i think this is not relate to next, but more into react world.. 

like seems : react + useeffect + socket + usestate + useref are not the 'best' partner ever :(  

i even follow here : 
https://socket.io/how-to/use-with-react

somewhere somehow it create another socket client instance, 
and snowball effect occour :(

**Comment by themegabyte** [comment]
There is a possible fix for this: https://stackoverflow.com/questions/74354724/socket-io-socket-listener-not-updating-in-nextjs-api

Please note, I don't think this is a good idea for production.

**Comment by NichuNaizam** [comment]
Any updates?
